### PR TITLE
fix order positions

### PR DESCRIPTION
**Pull request by patrickweh**
fix deleting a single position

add validation to prevent modifying locked orders orderpositions

add tests

### CODE REVIEW

File: resources/views/livewire/order/order-positions.blade.php [hunk 1]
```diff
@@ -100,7 +100,13 @@ class="pb-4"
             <x-slot:footer>
                 <div class="flex justify-between gap-x-4">
                     <div x-show="$wire.orderPosition.id">
-                        <x-button flat negative :label="__('Delete')" x-on:click="$wire.deleteOrderPosition(); close();" />
+                        <x-button
+                            flat
+                            negative
+                            :label="__('Delete')"
+                            wire:click="deleteOrderPosition(); close();"
+                            wire:flux-confirm.icon.error="{{  __('wire:confirm.delete', ['model' => __('Order Position')]) }}"
```
Comment (SirSplasch) [inline resources/views/livewire/order/order-positions.blade.php:108]: double whitespace before __()

File: resources/views/livewire/order/order-positions.blade.php [hunk 2]
```diff
@@ -100,7 +100,13 @@ class="pb-4"
             <x-slot:footer>
                 <div class="flex justify-between gap-x-4">
                     <div x-show="$wire.orderPosition.id">
-                        <x-button flat negative :label="__('Delete')" x-on:click="$wire.deleteOrderPosition(); close();" />
+                        <x-button
+                            flat
+                            negative
+                            :label="__('Delete')"
+                            wire:click="deleteOrderPosition(); close();"
```
Comment (SirSplasch) [inline resources/views/livewire/order/order-positions.blade.php:107]: wait this works? To call close(); afterwards?

File: src/Livewire/Order/OrderPositions.php [hunk 3]
```diff
@@ -425,6 +432,21 @@ public function deleteSelectedOrderPositions(): void
         $this->reset('selected');
     }
 
+    #[Renderless]
+    public function deleteOrderPosition(): void
```
Comment (SirSplasch) [inline src/Livewire/Order/OrderPositions.php:436]: why not return boolean, and handle this like every other delete?

**Review by SirSplasch — CHANGES_REQUESTED** [review]

File: resources/views/livewire/order/order-positions.blade.php (hunk 2 above)
Comment (patrickweh) [reply resources/views/livewire/order/order-positions.blade.php:107]: sure, wire:click is basically the same as x-on:click="$wire.doSomeThing", so you can add whatever javascript you want afterwards

**Review by SirSplasch — APPROVED** [review]

**Pull request merged**